velox/functions/prestosql/window/tests/NthValueTest.cpp

```diff
@@ -200,68 +193,79 @@ TEST_F(NthValueTest, offsetValues) {
+  assertWindowFunctionError(
+      {vectors},
+      "nth_value(c0, 5)",
+      overClause,
+      "rows between 0 preceding and current row",
+      "k in frame bounds must be at least 1");
+  assertWindowFunctionError(
+      {vectors},
+      "nth_value(c0, 5)",
+      overClause,
+      "rows between c2 preceding and current row",
+      "k in frame bounds must be at least 1");
 }
 
-TEST_F(NthValueTest, singlePartitionRangeFrames) {
-  auto vectors = makeSinglePartitionVectors(400);
+TEST_P(MultiNthValueTest, integerValues) {
+  testPrimitiveType(INTEGER());
+}
 
-  testWindowFunction(
-      {vectors}, "nth_value(c0, c2)", kFrameOverClauses, kRangeFrameClauses);
-  testWindowFunction(
-      {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRangeFrameClauses);
+TEST_P(MultiNthValueTest, tinyintValues) {
+  testPrimitiveType(TINYINT());
 }
 
-TEST_F(NthValueTest, multiInputRangeFrames) {
-  auto vectors = makeSinglePartitionVectors(200);
-  auto doubleVectors = {vectors, vectors};
+TEST_P(MultiNthValueTest, smallintValues) {
+  testPrimitiveType(SMALLINT());
+}
 
-  testWindowFunction(
-      doubleVectors,
-      "nth_value(c0, c2)",
-      kFrameOverClauses,
-      kRangeFrameClauses);
-  testWindowFunction(
-      {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRangeFrameClauses);
+TEST_P(MultiNthValueTest, bigintValues) {
+  testPrimitiveType(BIGINT());
 }
 
-TEST_F(NthValueTest, basicRowFrames) {
-  auto vectors = makeBasicVectors(50);
+TEST_P(MultiNthValueTest, realValues) {
+  testPrimitiveType(REAL());
+}
 
-  testWindowFunction(
-      {vectors}, "nth_value(c0, c2)", kFrameOverClauses, kRowsFrameClauses);
-  testWindowFunction(
-      {vectors}, "nth_value(c0, 1)", kFrameOverClauses, kRowsFrameClauses);
-  testWindowFunction(
-      {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRowsFrameClauses);
+TEST_P(MultiNthValueTest, doubleValues) {
+  testPrimitiveType(DOUBLE());
 }
 
-TEST_F(NthValueTest, singlePartitionRowFrames) {
-  auto vectors = makeSinglePartitionVectors(400);
+TEST_P(MultiNthValueTest, varcharValues) {
+  testPrimitiveType(VARCHAR());
+}
 
-  testWindowFunction(
-      {vectors}, "nth_value(c0, c2)", kFrameOverClauses, kRowsFrameClauses);
-  testWindowFunction(
-      {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRowsFrameClauses);
+TEST_P(MultiNthValueTest, varbinaryValues) {
+  testPrimitiveType(VARBINARY());
 }
 
-TEST_F(NthValueTest, multiInputRowFrames) {
-  auto vectors = makeSinglePartitionVectors(200);
-  auto doubleVectors = {vectors, vectors};
+TEST_P(MultiNthValueTest, timestampValues) {
+  testPrimitiveType(TIMESTAMP());
+}
 
-  testWindowFunction(
-      doubleVectors, "nth_value(c0, c2)", kFrameOverClauses, kRowsFrameClauses);
-  testWindowFunction(
-      {vectors}, "nth_value(c0, 5)", kFrameOverClauses, kRowsFrameClauses);
+TEST_P(MultiNthValueTest, dateValues) {
+  testPrimitiveType(DATE());
 }
 
+VELOX_INSTANTIATE_TEST_SUITE_P(
```

Comment:
What's the difference between these 2 tests? MultiFrameNthValueTest vs MultiNthValueTest?